### PR TITLE
Add ApplyUsingTree that represents context parameter application

### PR DESCRIPTION
**Pull request by tanishiking**
Close https://github.com/scalameta/scalameta/issues/2460

Added Scala's `Synthetic` part:
Screen Shot 2021-09-05 at 16 40 24

### CODE REVIEW

**Comment by tanishiking** [comment]
FYI @bishabosha

File: docs/semanticdb/specification.md [hunk 1]
```diff
@@ -1038,6 +1038,7 @@ message Tree {
     OriginalTree original_tree = 6;
     SelectTree select_tree = 7;
     TypeApplyTree type_apply_tree = 8;
+    ApllyUsingTree apply_using_tree = 9;
```
Comment (tgodzik) [inline docs/semanticdb/specification.md:1041]:
```suggestion
    ApllyContextTree apply_using_tree = 9;
```
maybe? `Using` feels a bit too general. What do you think @olafurpg ?

Comment (kitbellew) [inline docs/semanticdb/specification.md:1041]: didn't read the entire pr, but isn't there a typo here?

**Comment by tanishiking** [comment]
Closing as I don't have a plan to go with this change in the neqr future

**Pull request closed without merging**